pkg/apis/apps/v1alpha1/servicebinding_types.go

```diff
@@ -18,96 +18,92 @@ type ServiceBindingRequestSpec struct {
+	// This would be made a required field after 'services'
 	// is removed.
 	// +optional
-	BackingServiceSelectors *[]BackingServiceSelector `json:"backingServiceSelectors,omitempty"`
+	Services []Service `json:"services,omitempty"`
 
-	// ApplicationSelector is used to identify the application connecting to the
+	// Application is used to identify the application connecting to the
 	// backing service operator.
 	// +optional
-	ApplicationSelector ApplicationSelector `json:"applicationSelector"`
+	Application Application `json:"application"`
 
 	// DetectBindingResources is flag used to bind all non-bindable variables from
 	// different subresources owned by backing operator CR.
 	// +optional
 	DetectBindingResources bool `json:"detectBindingResources,omitempty"`
 }
 
-// ServiceBindingRequestStatus defines the observed state of ServiceBindingRequest
+// ServiceBindingStatus defines the observed state of ServiceBindingStatus
 // +k8s:openapi-gen=true
-type ServiceBindingRequestStatus struct {
+type ServiceBindingStatus struct {
 	// Conditions describes the state of the operator's reconciliation functionality.
 	Conditions []conditionsv1.Condition `json:"conditions"`
 	// Secret is the name of the intermediate secret
-	Secret string `json:"secret"`
+	Secret corev1.LocalObjectReference `json:"secret"`
 	// ApplicationObjects contains all the application objects filtered by label
 	// +optional
 	Applications []BoundApplication `json:"applications,omitempty"`
 }
 
-// BackingServiceSelector defines the selector based on resource name, version, and resource kind
-type BackingServiceSelector struct {
-	metav1.GroupVersionKind `json:",inline"`
-	ResourceRef             string `json:"resourceRef"`
+// Service defines the selector based on resource name, version, and resource kind
+type Service struct {
+	metav1.GroupVersionKind     `json:",inline"`
+	corev1.LocalObjectReference `json:",inline"`
+	EnvVarPrefix                *string `json:"envVarPrefix,omitempty"`
+
 	// +optional
-	Namespace    *string `json:"namespace,omitempty"`
-	EnvVarPrefix *string `json:"envVarPrefix,omitempty"`
+	// This is en EXPERIMENTAL feature till
+	// we support Subject Access Reviews.
+	Namespace *string `json:"namespace,omitempty"`
 }
 
 // BoundApplication defines the application workloads to which the binding secret has
-// injected.
+// injected. This is used in the status subresource.
 type BoundApplication struct {
 	metav1.GroupVersionKind     `json:",inline"`
 	corev1.LocalObjectReference `json:",inline"`
 }
 
-// ApplicationSelector defines the selector based on labels and GVR
-type ApplicationSelector struct {
+// Application defines the selector based on labels and GVR
+type Application struct {
 	// +optional
 	LabelSelector               *metav1.LabelSelector `json:"labelSelector,omitempty"`
 	metav1.GroupVersionResource `json:",inline"`
-	ResourceRef                 string `json:"resourceRef,omitempty"`
+	corev1.LocalObjectReference `json:",inline"`
```

Comment:
Isn't this `// +optional` as well? In case when `LabelSelector` is being used.

Reply:
Yeah, the content of the `struct` is actually optional
```
LocalObjectReference struct {
    // Name of the referent.
    // More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
    // TODO: Add other useful fields. apiVersion, kind, uid?
    // +optional
    Name string `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`
}
```

We could bubble up the information at this `Application` struct level too though.